hive/src/main/java/com/netease/arctic/hive/write/AdaptHiveOutputFileFactory.java

```diff
@@ -0,0 +1,111 @@
+ *    ${table_location}
+ *            -| change
+ *            -| base
+ *            -| hive_data
+ *                 -| ${partition_name1}
+ *                 -| ${partition_name2}
+ *                            -| txid=${txid}
+ *
+ * For Keyed adapt hive without partitions the dir construct is :
+ *    ${table_location}
+ *            -| change
+ *            -| base
+ *            -| hive_data
+ *                  -| ${timestamp_uuid}
+ *
+ * For UnKeyed adapt hive with partitions the dir construct is :
+ *    ${table_location}
+ *            -| base
+ *            -| hive_data
+ *                 -| ${partition_name1}
+ *                 -| ${partition_name2}
+ *                            -| txid=${txid}
+ *
+ * For Keyed adapt hive without partitions the dir construct is :
+ *    ${table_location}
+ *            -| base
+ *            -| hive_data
+ *                  -| ${timestamp_uuid}
+ */
+public class AdaptHiveOutputFileFactory implements OutputFileFactory {
+
+  private final String baseLocation;
+  private final PartitionSpec partitionSpec;
+  private final FileFormat format;
+  private final ArcticFileIO io;
+  private final EncryptionManager encryptionManager;
+  private final int partitionId;
+  private final long taskId;
+  private final long transactionId;
+
+  private final String unPartitionTableTmpDir = System.currentTimeMillis() + "_" + UUID.randomUUID();
+
+  private final String partitionTableTmpDir;
+
+  private final AtomicLong fileCount = new AtomicLong(0);
+
+  public AdaptHiveOutputFileFactory(
+      String baseLocation,
+      PartitionSpec partitionSpec,
+      FileFormat format,
+      ArcticFileIO io,
+      EncryptionManager encryptionManager,
+      int partitionId,
+      long taskId,
+      long transactionId) {
+    this.baseLocation = baseLocation;
+    this.partitionSpec = partitionSpec;
+    this.format = format;
+    this.io = io;
+    this.encryptionManager = encryptionManager;
+    this.partitionId = partitionId;
+    this.taskId = taskId;
+    this.transactionId = transactionId;
+    partitionTableTmpDir = "txid=" + transactionId;
+  }
+
+
+  private String generateFilename(TaskWriterKey key) {
+    if (key.getTreeNode() != null) {
+      return format.addExtension(
+          String.format("%d-%s-%d-%05d-%d-%010d", key.getTreeNode().getId(), key.getFileType().shortName(),
+              transactionId, partitionId, taskId, fileCount.incrementAndGet()));
+    } else {
+      return format.addExtension(
+          String.format("%s-%05d-%d-%010d",
+              key.getFileType().shortName(), partitionId, taskId, fileCount.incrementAndGet()));
+    }
+  }
+
+  private String fileLocation(StructLike partitionData, String fileName) {
+    if (partitionSpec.isUnpartitioned()) {
+      return String.format("%s/%s/%s", baseLocation, unPartitionTableTmpDir, fileName);
+    } else {
+      return String.format("%s/%s/%s/%s", baseLocation, partitionSpec.partitionToPath(partitionData),
```

Comment:
whatif  partitioned unkeyed table

Reply:
partition rule  has not relation with key or unkey

Reply:
I get you